spec.html

```diff
@@ -42606,6 +42585,7 @@ <h1>
         </dl>
         <emu-alg>
           1. Let _newTypedArray_ be ? Construct(_constructor_, _argumentList_).
+          1. Assert: _newTypedArray_ has all the internal slots mentioned in <emu-xref href="#sec-properties-of-typedarray-instances" title></emu-xref>.
```

Comment:
I don't see why this would hold? The `_constructor_` argument is generically "a constructor", not specifically a TA constructor, and indeed in e.g. [%TypedArray%.from](https://tc39.es/ecma262/multipage/indexed-collections.html#sec-%typedarray%.from) step 6.c this AO is invoked without any validation of the `_constructor_` argument except that it is a constructor.

The old asserts in the callees of this AO were justified because [ValidateTypedArray](https://tc39.es/ecma262/multipage/indexed-collections.html#sec-validatetypedarray) does a runtime check for the  [[TypedArrayName]] slot, which implies all the others.

So I think this is OK if you move it down one line.

Reply:
Yeah, nice catch about being off by one line. Fixed.